flink-connector-aws-kinesis-streams/src/main/java/org/apache/flink/connector/kinesis/sink/KinesisStreamsSinkWriter.java

```diff
@@ -253,4 +263,26 @@ private boolean isRetryable(Throwable err) {
 
         return true;
     }
+
+    private String lookupStreamArn(final String streamName) {
+        if (streamArn.get() == null) {
+            try {
+                final String arn =
+                        kinesisClient
+                                .describeStream(
+                                        DescribeStreamRequest.builder()
+                                                .streamName(streamName)
+                                                .build())
+                                .get()
+                                .streamDescription()
+                                .streamARN();
+                streamArn.set(arn);
+                return arn;
+            } catch (InterruptedException | ExecutionException e) {
+                throw new KinesisStreamsException(
```

Comment:
What happens if we get throttled? A Flink app with large parallelism could easily breach the 10 TPS on DescribeStream